Docs: breadcrumb usage docs (#1206)
carbon-design-system/carbon-website: src/pages/components/breadcrumb/usage.mdx

<AnchorLinks>

<AnchorLink>Overview</AnchorLink>
<AnchorLink>Live demo</AnchorLink>
<AnchorLink>Formatting</AnchorLink>
<AnchorLink>Content</AnchorLink>
<AnchorLink>Behaviors</AnchorLink>
<AnchorLink>Modifiers</AnchorLink>
<AnchorLink>Related</AnchorLink>
<AnchorLink>Feedback</AnchorLink>

</AnchorLinks>

## Overview

Breadcrumbs show users their current location relative to the information architecture and enable them to quickly move up to a parent level or previous step.

### When to use

> Breadcrumbs are effective in products and experiences that have a large amount of content organized in a hierarchy of more than two levels. They take up little space but still provide context for where in the navigation hierarchy the user is.

Reviewer: ```suggestion
Breadcrumbs are effective in products and experiences that have a large amount of content organized in a hierarchy of more than two levels. They take up little space but still provide context for the user's place in the navigation hierarchy.
```

### When not to use

> Breadcrumbs are always treated as secondary and should never entirely replace the primary navigation. They shouldn't be used for products that have single-level navigation because they create unnecessary clutter.

Reviewer: ```suggestion
Breadcrumbs are always treated as secondary and should never entirely replace the primary navigation. They shouldn't be used for products that have single level navigation because they create unnecessary clutter.
```

> If you are taking users through a multi-step process use a [progress indicator](/components/progress-indicator/usage/) instead.

Reviewer: ```suggestion
If you are taking users through a multistep process use a [progress indicator](/components/progress-indicator/usage/) instead.
```

`}</ComponentVariant>
</ComponentDemo>

## Formatting

### Anatomy

<Row>
<Column colLg={8}>

![Breadcrumb anatomy example](images/breadcrumb-anatomy.png)

</Column>
</Row>

1. **Page link:** Directs users to the parent-level page.
2. **Separator:** Clearly distinguishes between each page.

### Placement

Breadcrumbs are placed in the top left portion of the page. They sit underneath the header and navigation, but above the page title.

<Row>
<Column colLg={8}>

![Breadcrumb placement in a UI](images/breadcrumb-placement.png)

</Column>
</Row>

## Content

### Main elements

#### Page link

- Each page link should be short and clearly reflect the location or entity it links to.

> - List pages starting with the highest-level parent page and move deeper down the information architecture as the breadcrumb trail progresses.

Reviewer: ```suggestion
- Start with the highest level parent page and move deeper into the information architecture as the breadcrumb trail progresses.
```

#### Mouse

Users can trigger an item by clicking on a breadcrumb page link. The separators between page links are not interactive.

#### Keyboard

Users can trigger a breadcrumb link by pressing `Enter` while the link has focus. For additional keyboard interactions, see the [accessibility tab](https://www.carbondesignsystem.com/components/breadcrumb/accessibility).

Users can navigate between breadcrumb links by pressing `Tab` and `Shift-Tab`.

#### Screen readers

VoiceOver: Users can trigger a breadcrumb link by pressing `Enter` or `Control-Option-Space`. Users can navigate between breadcrumb links by pressing `Tab` and `Shift-Tab`.

Jaws: Users can trigger a breadcrumb link by pressing `Enter`. Users can navigate between breadcrumb links by pressing the `Up` and `Down` arrow keys.

NVDA: Users can trigger a breadcrumb link by pressing `Enter`. Users can navigate between breadcrumb links by pressing the `Up` and `Down` arrow keys.

For additional information, see [screen reader tests](https://www.carbondesignsystem.com/components/breadcrumb/accessibility#macos-screen-reader-tests).

## Modifiers

> By default Carbon breadcrumb trails should not include the current page. If a page doesn't have a title or the current page is not clear, it can be included in the breadcrumb trail. If the current page is included in a breadcrumb trail, it is always the last page link listed and is not an interactive link.

Reviewer: ```suggestion
By default, Carbon breadcrumb trails should not include the current page. If a page doesn't have a title or the current page is not clear, it can be included in the breadcrumb trail. If the current page is included in a breadcrumb trail, it is always the last page link listed and is not an interactive link.
```

<Row>
<Column colLg={8}>

![Breadcrumb showing current page](images/breadcrumb-current-page.png)

</Column>
</Row>

## Related

- [Global header](/patterns/global-header)
- [Progress indicator](/components/progress-indicator/usage/)

> - [UI shell header](/components/UI-shell-header/code)

Reviewer: ```suggestion
- [UI shell header](/components/UI-shell-header/usage)
```
